test/functional/terminal/tui_spec.lua

```diff
@@ -871,14 +871,22 @@ describe('TUI background color', function()
     screen:expect{any='did OptionSet, yay!'}
   end)
 
+  local last_bg
   local function assert_bg(colorspace, color, bg)
     it('handles '..color..' as '..bg, function()
+
+      -- Ensure the opposite of the expected bg is active.
+      if bg == last_bg then
+        command('set bg=' .. (bg == 'dark' and 'light' or 'dark'))
```

Comment:
similarly, I think this should use `feed_data(':set bg=' ... '\n')`

Reply:
Just for reference: this does not work due to the option then being considered to be set already (https://github.com/blueyed/neovim/blob/d2e34b3ee048dd5333516cc0fe1a7819412f4771/src/nvim/tui/input.c#L374).